packages/product/src/repositories/product.ts

```diff
@@ -121,7 +125,34 @@ export class ProductRepository extends AbstractBaseRepository<Product> {
       return manager.create(Product, product)
     })
 
-    await manager.persist(products)
+    await manager.persistAndFlush(products)
+
+    return products
+  }
+
+  async update(
+    data: WithRequiredProperty<ProductServiceTypes.UpdateProductDTO, "id">[],
+    context: Context = {}
+  ): Promise<Product[]> {
+    const manager = (context.transactionManager ??
+      this.manager_) as SqlEntityManager
+
+    const products = await Promise.all(
+      data.map(async (updateData) => {
+        if (!isDefined(updateData.id)) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_FOUND,
+            `Cannot update product without id`
+          )
+        }
+
+        const product = await manager.findOneOrFail(Product, updateData.id);
+
+        return manager.assign(product, updateData, { updateNestedEntities: true, mergeObjects: true })
+      })
+    )
```

Comment:
todo: this is not a real bulk here, I would suggest to find all the product by id and check if some are missing and in that case throw. then assign the data base on the product.id and updateData.id and the assign and persist all